sklearn/tree/_criterion.pyx

```diff
@@ -963,6 +964,99 @@ cdef class MSE(RegressionCriterion):
         impurity_right[0] /= self.n_outputs
 
 
+cdef class MAE(RegressionCriterion):
+    """Mean absolute error impurity criterion.
+
+        MAE = mae_left + mae_right
+    """
+    cdef double node_impurity(self) nogil:
+        """Evaluate the impurity of the current node, i.e. the impurity of
+           samples[start:end]."""
+
+        cdef double* sum_total = self.sum_total
+        cdef double impurity
+        cdef SIZE_t k
+
+        impurity = self.sq_sum_total / self.weighted_n_node_samples
```

Comment:
Can you explain why do you use `sq_sum_total`? (I would have expected `sum_total`)